src/components/ReportUpload/CounterUpload/CounterUpload.test.js

```diff
@@ -44,7 +42,13 @@ const uploadFile = async ({ mockFile, expectedError, mockHandler }) => {
   await userEvent.click(saveButton);
 
   if (expectedError) {
-    await waitFor(() => expect(onFail).toHaveBeenCalledWith(expectedError));
+    await waitFor(() => {
+      expect(screen.getByText(expectedError)).toBeInTheDocument();
+    });
+    if (expectedDetails) {
+      const details = screen.getByText(/more information/i).closest('details');
+      expect(within(details).getByText(expectedDetails)).toBeInTheDocument();
+    }
```

Comment:
Given the recent changes, the helper function should verify that the details section is not present when no details are expected.


```suggestion
    if (expectedDetails) {
      const details = screen.getByText(/more information/i).closest('details');
      expect(within(details).getByText(expectedDetails)).toBeInTheDocument();
    } else {
      expect(screen.queryByText(/more information/i)).not.toBeInTheDocument();
    }
```

Reply:
The else case is not working for the test cases without error code but rendering the accordion
`CounterUpload › upload scenario: error without code and message properties (err is undefined)`
`CounterUpload › upload scenario: error 404 without response body`

Reply:
@alb3rtino I made some more changes, based on your suggestion.